client/proxy.go

```diff
@@ -0,0 +1,1055 @@
+		}
+
+		err = ioutil.WriteFile(proxyDir+"/config/qdrouterd.json", []byte(mc), 0644)
+		if err != nil {
+			return fmt.Errorf("Failed to write config file: %w", err)
+		}
+
+		url, err := ioutil.ReadFile(proxyDir + "/config/url.txt")
+		agent, err := qdr.Connect(string(url), nil)
+		if err != nil {
+			return fmt.Errorf("qdr agent error: %w", err)
+		}
+		if err = agent.Delete("org.apache.qpid.dispatch.tcpConnector", proxyName+"-egress-"+address); err != nil {
+			return fmt.Errorf("Error removing tcp connector : %w", err)
+		}
+		agent.Close()
+	}
+
+	return nil
+}
+
+func (cli *VanClient) ProxyExpose(ctx context.Context, options types.ProxyExposeOptions) (string, error) {
+
+	proxyName, err := cli.ProxyInit(ctx, types.ProxyInitOptions{
+		Name:       options.ProxyName,
+		StartProxy: false,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// Note: expose implicitly creates the service
+	si, err := cli.ServiceInterfaceInspect(ctx, options.Egress.Address)
+	if err != nil {
+		return "", fmt.Errorf("Failed to retrieve service: %w", err)
+	}
+
+	if si == nil {
+		servicePort, err := strconv.Atoi(options.Egress.Port)
+		if err != nil {
+			return "", fmt.Errorf("%s is not a valid port", options.Egress.Port)
+		}
+		err = cli.ServiceInterfaceCreate(context.Background(), &types.ServiceInterface{
+			Address:  options.Egress.Address,
+			Protocol: options.Egress.Protocol,
+			Port:     servicePort,
+		})
+		if err != nil {
+			return "", fmt.Errorf("Unabled to create service: %w", err)
+		}
+	}
+
+	err = cli.ProxyBind(ctx, proxyName, options.Egress)
+	if err != nil {
+		return proxyName, err
+	}
+
+	err = cli.proxyStart(ctx, proxyName)
+	if err != nil {
+		return proxyName, err
+	}
+
+	return proxyName, nil
+}
+
+func (cli *VanClient) ProxyUnexpose(ctx context.Context, proxyName string, address string) error {
+	// Note: no need to unbind as proxy is being deleted
+	// Note: unexpose implicitly removes the cluster service
+	si, err := cli.ServiceInterfaceInspect(ctx, address)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve service: %w", err)
+	}
+
+	// todo: only remove service if not used, is this necessary
+	if si != nil && len(si.Targets) == 0 && si.Origin == "" {
+		err := cli.ServiceInterfaceRemove(ctx, address)
+		if err != nil {
+			return fmt.Errorf("Failed to removes service: %w", err)
+		}
+	}
+
+	// Note: unexpose will stop and remove proxy independent of bridge configuration
+	err = cli.proxyStop(ctx, proxyName)
+	if err != nil {
+		return err
+	}
+
+	err = cli.ProxyRemove(ctx, proxyName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cli *VanClient) ProxyInspect(ctx context.Context, proxyName string) (*types.ProxyInspectResponse, error) {
+	proxyDir := getDataHome() + "/skupper/" + proxyName
+
+	_, err := getRootObject(cli)
+	if err != nil {
+		return nil, fmt.Errorf("Skupper not initialized in %s", cli.Namespace)
+	}
+
+	proxyVersion, err := exec.Command("qdrouterd", "-v").Output()
```

Comment:
Minor point: Could also get the version via the agent (infact agent.local.Version would gve it to you but not at present accessible outside the package. Could add a `func (a *Agent) getLocalRouterVersion() string` though. Not essential, not necessarily even better, just mentioning in passing.